src/solid_3D_ele/4C_solid_3D_ele_calc_lib.hpp

```diff
@@ -856,6 +904,50 @@ namespace Discret::Elements
         integration_fac, Bop, Core::LinAlg::make_stress_like_voigt_view(stress.pk2_), 1.);
   }
 
+  template <Core::FE::CellType celltype>
+  void add_internal_force_vector(const JacobianMapping<celltype>& jacobian_mapping,
+      const Core::LinAlg::Tensor<double, Core::FE::dim<celltype>, Core::FE::dim<celltype>>& F,
+      const Core::LinAlg::SymmetricTensor<double, Core::FE::dim<celltype>, Core::FE::dim<celltype>>&
+          pk2,
```

Comment:
You could consider using stronger types here. It seems very easy to mess up by giving the wrong tensors. E.g. with

```
struct DefGrad
{
const Core::LinAlg::Tensor<double, Core::FE::dim<celltype>, Core::FE::dim<celltype>>& f;
};
```
you would change this to

```suggestion
      const DefGrad& F,
      const Core::LinAlg::SymmetricTensor<double, Core::FE::dim<celltype>, Core::FE::dim<celltype>>&
          pk2,
```

and at the call site use `add_internal_force_vector(..., DefGrad{F}, ...)`.

 I am not sure how general this function is. If it is only an internal helper, this may not be as useful.